src/Workspace/WorkspaceItem.elm

```diff
@@ -54,6 +54,16 @@ type Item
     | AbilityConstructorItem AbilityConstructorDetail
 
 
+{-| WorkspaceItem doesn't manage state itself, but has a limit set of actions
+-}
+type Msg
+    = Close Reference
+    | OpenReference Reference Reference
+    | UpdateZoom Reference Zoom
+    | ToggleDocFold Reference Doc.FoldId
+    | ChangePerspectiveToNamespace FQN
```

Comment:
Since I was needing to add a handler for changing perspective, I decided to refactor this to be a Msg from WorkspaceItem instead of its `view` function taking in 5 callbacks.